### PR TITLE
Update register page with custom errors

### PR DESCRIPTION
**Pull request by wariofdz**
### Added:

- Added new function `getCreateUserErrorMessage.ts` and his test, to get a error message for create user.
- Added new file `createUserErrorMessages.ts` with error message for create user.
- Added new file `mapUseQueryHookResultV2.ts` and his test, to get a user created or a code of error.
- Added new function `isSerializableAppError.ts` and his test, to check the type of error.

### Changed: 

- Modified interface `UseRegisterData.ts` to add a new optional property `errorMessage`.
- Modified `package.json` to add the workspace `frontend-common`.
- Modified `useRegister.ts` and his test, to add new functionality with `mapUseQueryHookResultV2`.
- Modified `SerializableAppError.ts` and hist test, to add a `Kind` with a generic type.

### CODE REVIEW

File: packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        code: 'code-fixture',
+        message: 'message-fixture',
+        name: 'name-fixture',
+        stack: 'stack-fixture',
+      },
+      false,
+    ],
+    ['a non object', {}, false],
+  ])(
+    'having %s error',
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts:29]:
```suggestion
    'having %s value',
```

File: packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    [
+      'a SerializableAppError',
+      {
+        kind: AppErrorKind.contractViolation,
+        message: 'message-fixture',
+      },
+      true,
+    ],
+    [
+      'a SerializedError',
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts:18]: This case is too specific. We are writing an unit test here. I'm missing the following cases:
- A non object.
- `null` value.
- An object with no `kind`.
- An object with invalid `kind`.
- An object with valid kind and no message.

File: packages/frontend/web-ui/src/auth/hooks/useRegister.ts [hunk 3]
```diff
@@ -192,6 +200,7 @@ export const useRegister = (): [UseRegisterData, UseRegisterActions] => {
     } else {
       setRegisterData({
         form: {
+          errorMessage: '',
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/auth/hooks/useRegister.ts:203]:
```suggestion
          errorMessage: undefined,
```
An empty string error is still an error. We have no error from the endpoint at this point, right?

**Review by notaphplover — CHANGES_REQUESTED** [review]
It looks grear! I just added some minor suggestions.

File: packages/frontend/web-ui/src/auth/hooks/useRegister.ts (hunk 3 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/auth/hooks/useRegister.ts:203]: You're right! I've deleted this line.

File: packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts]: I've changed this description

File: packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts (hunk 2 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/common/helpers/isSerializableAppError.spec.ts]: I've deleted the case of SerializedError and I've added the cases that you commented.

**Review by notaphplover — APPROVED** [review]

**Pull request merged**